pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,54 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
```

Comment:
If we find we want this in the future a new ticket should be opened. There have been other comments surrounding this function as well. However, right now other functions also use the wait functions that already existed and still operate under certain expectations potentially.